hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/TestHoodieFileIndex.scala

```diff
@@ -383,6 +383,66 @@ class TestHoodieFileIndex extends HoodieSparkClientTestBase with ScalaAssertionS
     }
   }
 
+  /**
+   * This test mainly ensures all non-partition-prefix filter can be pushed successfully
+   */
+  @ParameterizedTest
+  @CsvSource(value = Array("true, false", "false, false", "true, true", "false, true"))
+  def testPartitionPruneWithMultiplePartitionColumnsWithComplexExpression(useMetadataTable: Boolean,
+                                                                          complexExpressionPushDown: Boolean): Unit = {
```

Comment:
It covers the scenario when either both `URL_ENCODE_PARTITIONING` and `HIVE_STYLE_PARTITIONING` are true or both are false. What if one is true and the other is false? It should fallback to regular listing by prefix right? Is it being tested somehwere?